src/childBridgeFork.ts

```diff
@@ -0,0 +1,224 @@
+/**
+ * This is a standalone script executed as a child process fork
+ */
+
+// registering node-source-map-support for typescript stack traces
+import "source-map-support/register"; 
+
+import { AccessoryPlugin, HomebridgeAPI, PlatformPlugin, PluginType } from "./api";
+import { Plugin } from "./plugin";
+import { PluginManager } from "./pluginManager";
+import { Logger } from "./logger";
+import { User } from "./user";
+import { HAPStorage } from "hap-nodejs";
+import {
+  ChildProcessMessageEventType,
+  ChildProcessMessageEvent,
+  ChildProcessLoadEventData,
+} from "./childBridgeService";
+import {
+  AccessoryConfig,
+  BridgeConfiguration,
+  BridgeOptions,
+  BridgeService,
+  HomebridgeConfig,
+  PlatformConfig,
+} from "./bridgeService";
+
+process.title = "homebridge: child bridge";
+
+export class ChildBridgeFork {
+  private bridgeService!: BridgeService;
+  private api!: HomebridgeAPI;
+  private pluginManager!: PluginManager;
+
+  private type!: PluginType;
+  private plugin!: Plugin;
+  private identifier!: string;
+  private pluginConfig!: PlatformConfig | AccessoryConfig;
+  private bridgeConfig!: BridgeConfiguration;
+  private bridgeOptions!: BridgeOptions;
+  private homebridgeConfig!: HomebridgeConfig;
+
+  constructor() {
+    // tell the parent process we are ready to accept plugin config
+    this.sendMessage(ChildProcessMessageEventType.READY);
+  }
+
+  sendMessage<T = unknown>(type: ChildProcessMessageEventType, data?: T): void {
+    if (process.send) {
```

Comment:
Out of interest, in what cases in the send method not available? And would it make sense to explicitly handle that case?

Reply:
Process.send is not available unless the script is a child process spawned by .fork(). This condition  prevents TypeScript complaining.